src/client/clouds.cpp

```diff
@@ -92,6 +92,77 @@ void Clouds::OnRegisterSceneNode()
 	ISceneNode::OnRegisterSceneNode();
 }
 
+// Permutation table
+static const int permutation[] = {
+	151,160,137,91,90,15,131,13,201,95,96,53,194,233,7,225,140,36,103,30,69,142,8,99,37,240,21,10,23,
```

Comment:
~~What's the origin of this magic spell? Is there a formula to recalculate it?~~
~~Also: `u8` might be a better choice in place of `int` to pack the data more efficiently.~~

Comment might be obsolete. See `PerlinNoise2D`.

Reply:
using noise.cpp was my goal, however It severely crashes performance. To the point where my whole PC deadlocks.

Reply:
If mapgens work for you, then the suggested functions should too. Keep your task manager visible while testing. Maybe you're provoking a memory leak, resulting in heavy RAM spamming, which would explain your "whole PC deadlock".